core/amber/src/main/scala/edu/uci/ics/texera/workflow/common/tuple/Tuple.java

```diff
@@ -597,14 +617,27 @@ public BuilderV2(Schema schema) {
 
         /**
          * The tuple argument here is expected to conform to the exact same schema as the
-         * the schema passed in the constructor. Failure to do so will result in an exception.
+         * the schema passed in the constructor. If it doesn't conform, an error will be thrown.
+         * If you wish to ignore attributes in the tuple that are not part of the output schema,
+         * then use {@link Tuple.BuilderV2#add(Tuple, boolean)} with second parameter as false
          */
         public BuilderV2 add(Tuple tuple) {
+            return add(tuple, true);
+        }
+
+        public BuilderV2 add(Tuple tuple, boolean isStrictSchemaMatch) {
             checkNotNull(tuple);
 
             for (int i = 0; i < tuple.size(); i++) {
-                // TODO There is scope here to be "loose" in the schema matching, so that we don't need a "remove" ever
-                add(tuple.getSchema().getAttributes().get(i), tuple.getFields().get(i));
+                Attribute attribute = tuple.getSchema().getAttributes().get(i);
+                // The isStrictSchemaMatch parameter toggles the ability to check exact schema matching.
```

Comment:
I thought this change was in the last PR. Why is it again shown here?

Reply:
There are changes from the other PR that I'm working on, here as well, as there is a dependency between the two. Once that PR is merged, I'll merge master to this and the change should disappear.